daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2269,7 +2333,7 @@ func NewDeviceSet(root string, doInit bool, options []string, uidMaps, gidMaps [
 		metaDataLoopbackSize:  defaultMetaDataLoopbackSize,
 		baseFsSize:            defaultBaseFsSize,
 		overrideUdevSyncCheck: defaultUdevSyncOverride,
-		filesystem:            "ext4",
+		defaultFilesystem:     determineDefaultFS(),
```

Comment:
In fact I think I can get rid of devices.defaultFilesystem. As devices.defaultFilesystem is consulted only at one place (during file system creation), I think I will not even store it in devices. Just save it in a local variable and discard it after that.